src/ngraph/runtime/gpu/cudnn_descriptors.hpp

```diff
@@ -135,6 +135,19 @@ namespace ngraph
                 }
             };
 
+            template <>
```

Comment:
Specializations can be moved to the cpp file. This makes the header much cleaner

Reply:
I believe these specializations actually need to be defined in the header because the template function build<T> requires their definition. The forward declaration of 
```
            template <typename T>
            struct cudnn_descriptor;
``` 
doesn't suffice in this case.

Reply:
```
cudnn_descriptors.hpp:56:41: error: implicit instantiation of undefined template 'ngraph::runtime::gpu::cudnn_descriptor<cudnnTensorStruct *>'
                        descriptor_() { cudnn_descriptor<T>::create(stored_descriptor); }
```